nemo_skills/pipeline/script.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from pathlib import Path
+
+import nemo_run as run
+import typer
+
+from nemo_skills.pipeline import add_task, check_if_mounted, get_cluster_config, get_generation_command, run_exp
+from nemo_skills.pipeline.app import app, typer_unpacker
+from nemo_skills.utils import setup_logging
+
+LOG = logging.getLogger(__file__)
+
+
+def get_cmd(module, extra_arguments):
+    cmd = f"python -m {module} "
+    cmd += f" {extra_arguments} "
+    cmd = (
+        f"export PYTHONPATH=$PYTHONPATH:/nemo_run/code && "
+        f"cd /nemo_run/code && "
+        # might be required if we are not hosting server ourselves
+        f"export NVIDIA_API_KEY={os.getenv('NVIDIA_API_KEY', '')} && "
+        f"export OPENAI_API_KEY={os.getenv('OPENAI_API_KEY', '')} && "
+        f"{cmd}"
+    )
+    return cmd
+
+
+@app.command(context_settings={"allow_extra_args": True, "ignore_unknown_options": True})
+@typer_unpacker
+def script(
+    ctx: typer.Context,
+    cluster: str = typer.Option(
+        None,
+        help="One of the configs inside config_dir or NEMO_SKILLS_CONFIG_DIR or ./cluster_configs. "
+        "Can also use NEMO_SKILLS_CONFIG instead of specifying as argument.",
+    ),
+    expname: str = typer.Option("script", help="Nemo run experiment name"),
+    partition: str = typer.Option(
+        None, help="Can specify if need interactive jobs or a specific non-default partition"
+    ),
+    module: str = typer.Option(
```

Comment:
what exactly is copied to `/nemo_run/code` in this case? whatever is in the directory that they invoked `ns` from?

Reply:
^doesn't seem like this could be the case due to how the `packager` works: https://github.com/Kipok/NeMo-Skills/blob/main/nemo_skills/pipeline/utils.py#L419-L460

It is generally packing up whatever code is in the NeMo-Skills directory that this is being run from (be it git or pip)--there doesn't seem to be anything packaging user scripts outside of the NeMo-Skills installation path: https://github.com/NVIDIA/NeMo-Run/blob/5eea9cb47df6c0d2d3a4b04822250aaa9d171084/src/nemo_run/core/execution/slurm.py#L556-L569

Reply:
note: it seems if I run this inside a git repository, THEN it copies the contents of the working directory

Reply:
added in b6b17d48f271124626c4a0b7561321f112dcec25